PWGJE/EMCALJetTasks/AliEmcalJetTask.cxx

```diff
@@ -372,8 +381,19 @@ Bool_t AliEmcalJetTask::GetSortedArray(Int_t indexes[], std::vector<fastjet::Pse
  */
 void AliEmcalJetTask::ExecOnce()
 {
+  
+  // If a constant artificial track efficiency is supplied, create a TF1 that is constant in pT
   if (fTrackEfficiency < 1.) {
-    if (gRandom) delete gRandom;
+    fTrackEfficiencyFunction = TF1("trackEfficiencyFunction", "[0]", 0., fMaxBinPt);
```

Comment:
It looks like it would be possible for a user to set a track efficiency function as well as a `fTrackEfficiency`. Since I think we want to avoid that behavior, can you check if fTrackEfficiencyFunction is already valid (I'm not quite sure the best way to accomplish this) and throw an error if it is?

Reply:
Done